From 17bf18ca31626469594f43956fbb4a243687a969 Mon Sep 17 00:00:00 2001
From: Harish Mahendrakar <harish.mahendrakar@ittiam.com>
Date: Thu, 25 May 2017 16:38:51 +0530
Subject: [PATCH 02/16] Added an out of bound check on u4_num_bufs in input
 argument

ps_dec_ip->s_out_buffer.u4_num_bufs was missing out of bound checks

Bug: 35430570
Change-Id: Ibbf9891a885f69e208107725e34e7217147b891e
(cherry picked from commit 8221313d58ad4ebe9875760f065d999928172d6e)
CVE-2017-0851
---
 decoder/ihevcd_decode.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/decoder/ihevcd_decode.c b/decoder/ihevcd_decode.c
index 7b82e84..04ad8f5 100644
--- a/decoder/ihevcd_decode.c
+++ b/decoder/ihevcd_decode.c
@@ -421,7 +421,8 @@ WORD32 ihevcd_decode(iv_obj_t *ps_codec_obj, void *pv_api_ip, void *pv_api_op)
     if(0 == ps_codec->i4_share_disp_buf && ps_codec->i4_header_mode == 0)
     {
         UWORD32 i;
-        if(ps_dec_ip->s_out_buffer.u4_num_bufs == 0)
+        if((ps_dec_ip->s_out_buffer.u4_num_bufs <= 0) ||
+           (ps_dec_ip->s_out_buffer.u4_num_bufs > IVD_VIDDEC_MAX_IO_BUFFERS))
         {
             ps_dec_op->u4_error_code |= 1 << IVD_UNSUPPORTEDPARAM;
             ps_dec_op->u4_error_code |= IVD_DISP_FRM_ZERO_OP_BUFS;
-- 
2.7.4

